frontend/model/contracts/group.js

```diff
@@ -417,7 +417,7 @@ DefineContract({
           console.error(`paymentUpdate: bad username ${meta.username} != ${payment.meta.username} != ${payment.data.username}`, { data, meta, hash })
           throw new Errors.GIErrorIgnoreAndBanIfGroup('paymentUpdate from bad user!')
         }
-        payment.history.push([meta.createdDate, data.updatedProperties])
+        payment.history.push([meta.createdDate, hash])
```

Comment:
Dumb question: How do I access a message by hash?

For example, when we need to access `history[0][1]`, it will retrieve the hash, but then would can we access the actual data?

Reply:
@sandrina-p not a dumb question at all!

Each message is stored by hash via `addMessageToDB()`  in `state.js`, which calls `'gi.db/log/addEntry'`, which in turn is defined in `shared/domains/gi/db.js`, and uses the hash as the key into IndexedDB.

So you would call `'gi.db/log/getEntry'` using the hash to retrieve it.

Reply:
Gotcha! In what situations/UI will we need to access a payment history?

Reply:
@sandrina-p it's possible something like that might need to be done when doing the details of a payment.

Similarly, we will need to use that API to show the complete payment history, or anything that involves digging up history of anything (e.g. chat messages, etc.), since we cannot store everything in Vuex memory all the time.

Reply:
It makes sense. Thanks!